python/ray/llm/_internal/batch/stages/base.py

```diff
@@ -0,0 +1,211 @@
+"""TBA"""
+import logging
+from enum import Enum, auto
+from typing import Any, Dict, AsyncIterator, List, Callable, Tuple
+
+import pyarrow
+from pydantic import BaseModel
+from ray.data.block import UserDefinedFunction
+
+logger = logging.getLogger(__name__)
+
+
+def wrap_preprocess(
+    fn: UserDefinedFunction,
+    input_column: str,
+    carry_over: bool,
+) -> Callable:
+    """Wrap the preprocess function, so that the output schema of the
+    preprocess is normalized to {input_column: ...} (with other input columns
+    if carry_over is True).
+
+    Args:
+        fn: The function to be applied.
+        input_column: The input column name.
+        carry_over: Whether to carry over the output column to the next stage.
+
+    Returns:
+        The wrapped function.
+    """
+
+    def _preprocess(row: dict[str, Any]) -> dict[str, Any]:
+        outputs = {input_column: fn(row)}
+
+        if carry_over:
+            if input_column in row:
+                row.pop(input_column)
+
+            # In the case of column name conflict,
+            # we overwrite the input column with the output column.
+            row.update(outputs)
+        return row
+
+    return _preprocess
+
+
+def wrap_postprocess(
+    fn: UserDefinedFunction,
+    input_column: str,
```

Comment:
by input_column here do you mean the output_column of the processor? If so, maybe we should pick better names so that the developer is not confused. Something like 

```
wrap_preprocessor(fn, processor_input_column, ...)
wrap_postprocessor(fn, processor_output_column, ...)
```